File: src/order_type/ocme.js
import React from "react";
import "semantic-ui-css/semantic.min.css";

class OCME extends React.Component{

    render(){
        return(
            <div className="-order-modals">
                {this.props.order_info["customer"]["billing_name"]} <br/>
                {this.props.order_info["customer"]["address"]}  <br/>
                <h2>{this.props.order_info["order_type"]}</h2>
                <strong>Order number:</strong> {this.props.order_info["order_number"]}<br/>
                <strong>Order time:</strong> {this.props.order_info["date_submitted"]} <br/>
                <strong>Phone:</strong> {this.props.order_info["customer"]["phone"]} <br/>
                <strong>E-mail:</strong> {this.props.order_info["customer"]["email"]} <br/>
                <strong>Suborder Number</strong> {this.props.order_info["suborder_number"]} <br/> <br/>

                <div className="-row">
                    <div className="-fourth"> Borough: </div>
                    <div className="-two-thirds">
                        {this.props.order_info["metadata"]["borough"]}
                    </div>
                </div>

                <div className="-row">
                    <div className="-fourth"> Date: </div>
                    <div className="-two-thirds">
                        {this.props.order_info["metadata"]["date"]}
                    </div>
                </div>

                <div className="-row">
                    <div className="-fourth"> First Name: </div>
                    <div className="-two-thirds">
                        {this.props.order_info["metadata"]["first_name"]}
                    </div>
                </div>


                <div className="-row">
                    <div className="-fourth"> Middle Name: </div>
                    <div className="-two-thirds">
                        {this.props.order_info["metadata"]["middle_name"] ? this.props.order_info["metadata"]["middle_name"] : ""}
                    </div>
                </div>

                <div className="-row">
                    <div className="-fourth"> Last Name: </div>
                    <div className="-two-thirds">
                        {this.props.order_info["metadata"]["last_name"]}
                    </div>
                </div>

                <div className="-row">
                    <div className="-fourth"> Age: </div>
                    <div className="-two-thirds">
                        {this.props.order_info["metadata"]["age"]  ? this.props.order_info["metadata"]["age"] : ""}
                    </div>
                </div>

                <div className="-row">
                    <div className="-fourth"> Certificate Number: </div>
                    <div className="-two-thirds">
                        {this.props.order_info["metadata"]["certificate_number"] ? this.props.order_info["metadata"]["certificate_number"] : ""}
                    </div>
                </div>

                <div className="-row">
                    <div className="-fourth"> Number of Copies: </div>
                    <div className="-two-thirds"> {this.props.order_info["metadata"]["num_copies"]}</div>
                </div>

                <div className="-row">
                    <div className="-fourth"> Raised Seal: </div>
                    <div className="-two-thirds"> {this.props.order_info["metadata"]["raised_seal"] ? "Yes" : "No"}</div>
                </div>

                {this.props.order_info["metadata"]["raised_seal"] &&
                    <div className="-row">
                        <div className="-fourth"> Raised Seal Copies: </div>
                        <div className="-two-thirds">
                            {this.props.order_info["metadata"]["raised_seal_copies"] ? this.props.order_info["metadata"]["raised_seal_copies"] : ""}
                        </div>
                    </div>
                }

                <div className="-row">
                    <div className="-fourth"> Delivery Method: </div>
                    <div className="-two-thirds"> {this.props.order_info["metadata"]["delivery_method"]} </div>
                </div>

                {this.props.order_info["metadata"]["contact_number"] &&
                    <div className="-row">
                        <div className="-fourth"> Contact Number: </div>
                        <div className="-two-thirds">
                            {this.props.order_info["metadata"]["contact_number"]}
                        </div>
                    </div>
                }

                {this.props.order_info["metadata"]["contact_email"] &&
                    <div className="-row">
                        <div className="-fourth"> Contact Email: </div>
                        <div className="-two-thirds">
                            {this.props.order_info["metadata"]["contact_email"]}
                        </div>
                    </div>
                }
            </div>
        )
    };
}


export default OCME;

File: app/models/ocme.py
from sqlalchemy.dialects import postgresql

from app import db
from app.constants import borough, delivery_method


class OCME(db.Model):
    """
    id
    borough
    date
    first_name
    middle_name
    last_name
    age
    certificate_number
    num_copies
    raised_seal
    delivery_method
    """
    __tablename__ = "ocme"
    id = db.Column(db.Integer, primary_key=True)
    borough = db.Column(
        postgresql.ENUM(
            borough.BRONX,
            borough.MANHATTAN,
            borough.STATEN_ISLAND,
            borough.BROOKLYN,
            borough.QUEENS,
            name="borough",
            create_type=False
        ), nullable=False)
    date = db.Column(db.Date, nullable=False)
    first_name = db.Column(db.String(40), nullable=False)
    middle_name = db.Column(db.String(40), nullable=True)
    last_name = db.Column(db.String(25), nullable=False)
    age = db.Column(db.String(3), nullable=True)
    certificate_number = db.Column(db.String(40), nullable=True)
    num_copies = db.Column(db.String(1), nullable=False)
    raised_seal = db.Column(db.Boolean, nullable=False)
    raised_seal_copies = db.Column(db.String(1), nullable=True)
    delivery_method = db.Column(
        postgresql.ENUM(
            delivery_method.MAIL,
            delivery_method.EMAIL,
            delivery_method.PICKUP,
            name="delivery_method",
            create_type=False,
        ), nullable=False)
    contact_number = db.Column(db.String(64), nullable=True)
    contact_email = db.Column(db.String(256), nullable=True)
    suborder_number = db.Column(db.String(32), db.ForeignKey('suborders.id'), nullable=False)

    def __init__(
            self,
            borough,
            date,
            first_name,
            last_name,
            num_copies,
            raised_seal,
            delivery_method,
            suborder_number,
            middle_name=None,
            age=None,
            certificate_number=None,
            raised_seal_copies=None,
            contact_number=None,
            contact_email=None,
    ):
        self.borough = borough
        self.date = date
        self.first_name = first_name
        self.middle_name = middle_name
        self.last_name = last_name
        self.age = age
        self.certificate_number = certificate_number
        self.num_copies = num_copies
        self.raised_seal = raised_seal
        self.raised_seal_copies = raised_seal_copies
        self.delivery_method = delivery_method
        self.contact_number = contact_number
        self.contact_email = contact_email
        self.suborder_number = suborder_number

    @property
    def serialize(self):
        return {
            "id": self.id,
            "borough": self.borough,
            "date": self.date,
            "first_name": self.first_name,
            "middle_name": self.middle_name,
            "last_name": self.last_name,
            "age": self.age,
            "certificate_number": self.certificate_number,
            "num_copies": self.num_copies,
            "raised_seal": self.raised_seal,
            "raised_seal_copies": self.raised_seal_copies,
            "delivery_method": self.delivery_method,
            "contact_number": self.contact_number,
            "contact_email": self.contact_email,
            "suborder_number": self.suborder_number
        }
